File: cpp/GameProgress/GameStatContainer.c
// Type: GameProgress.GameStatContainer
// Ghidra (System V/gcc) decompilation, IL2CPP structs + signatures applied.
// ---- AoTTG2 cross-reference ----
// Update status: unchanged since the previous dump
// C# structure: source/csharp/Scripts/GameProgress/GameStatContainer.cs
// Prior real C# source (older reference): Assets/Scripts/GameProgress/GameStatContainer.cs
// --------------------------------

// GameProgress.GameStatContainer$$.ctor
// il2cpp: void GameProgress_GameStatContainer___ctor (GameProgress_GameStatContainer_o* __this, const MethodInfo* method);
// 0x453d5f0

void GameProgress_GameStatContainer___ctor(GameProgress_GameStatContainer_o *__this,MethodInfo *method)

{
  Settings_IntSetting_o *pSVar1;
  Settings_FloatSetting_o *pSVar2;
  GameProgress_DamageSetting_o *__this_00;
  
  if (g_data_057aef23 == '\0') {
    il2cpp_runtime_helper_023445d0(&TypeInfo_DamageSetting);
    il2cpp_runtime_helper_023445d0(&TypeInfo_FloatSetting);
    il2cpp_runtime_helper_023445d0(&TypeInfo_IntSetting);
    g_data_057aef23 = '\x01';
  }
  pSVar1 = (Settings_IntSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_IntSetting);
  Settings_IntSetting___ctor_40f39a0(pSVar1,1,-0x80000000,0x7fffffff,(MethodInfo *)0x0);
  (__this->fields).Level = pSVar1;
  il2cpp_runtime_helper_022b4080(&(__this->fields).Level);
  pSVar1 = (Settings_IntSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_IntSetting);
  Settings_IntSetting___ctor_40f39a0(pSVar1,0,-0x80000000,0x7fffffff,(MethodInfo *)0x0);
  (__this->fields).Exp = pSVar1;
  il2cpp_runtime_helper_022b4080(&(__this->fields).Exp);
  pSVar2 = (Settings_FloatSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_FloatSetting);
  Settings_FloatSetting___ctor_40f3a30(pSVar2,0.0,-3.4028235e+38,3.4028235e+38,(MethodInfo *)0x0);
  (__this->fields).PlayTime = pSVar2;
  il2cpp_runtime_helper_022b4080(&(__this->fields).PlayTime);
  pSVar2 = (Settings_FloatSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_FloatSetting);
  Settings_FloatSetting___ctor_40f3a30(pSVar2,0.0,0.0,100000.0,(MethodInfo *)0x0);
  (__this->fields).HighestSpeed = pSVar2;
  il2cpp_runtime_helper_022b4080(&(__this->fields).HighestSpeed);
  pSVar1 = (Settings_IntSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_IntSetting);
  Settings_IntSetting___ctor_40f39a0(pSVar1,0,-0x80000000,0x7fffffff,(MethodInfo *)0x0);
  (__this->fields).TitansKilledTotal = pSVar1;
  il2cpp_runtime_helper_022b4080(&(__this->fields).TitansKilledTotal);
  pSVar1 = (Settings_IntSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_IntSetting);
  Settings_IntSetting___ctor_40f39a0(pSVar1,0,-0x80000000,0x7fffffff,(MethodInfo *)0x0);
  (__this->fields).TitansKilledBlade = pSVar1;
  il2cpp_runtime_helper_022b4080(&(__this->fields).TitansKilledBlade);
  pSVar1 = (Settings_IntSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_IntSetting);
  Settings_IntSetting___ctor_40f39a0(pSVar1,0,-0x80000000,0x7fffffff,(MethodInfo *)0x0);
  (__this->fields).TitansKilledAHSS = pSVar1;
  il2cpp_runtime_helper_022b4080(&(__this->fields).TitansKilledAHSS);
  pSVar1 = (Settings_IntSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_IntSetting);
  Settings_IntSetting___ctor_40f39a0(pSVar1,0,-0x80000000,0x7fffffff,(MethodInfo *)0x0);
  (__this->fields).TitansKilledThunderspear = pSVar1;
  il2cpp_runtime_helper_022b4080(&(__this->fields).TitansKilledThunderspear);
  pSVar1 = (Settings_IntSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_IntSetting);
  Settings_IntSetting___ctor_40f39a0(pSVar1,0,-0x80000000,0x7fffffff,(MethodInfo *)0x0);
  (__this->fields).TitansKilledAPG = pSVar1;
  il2cpp_runtime_helper_022b4080(&(__this->fields).TitansKilledAPG);
  pSVar1 = (Settings_IntSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_IntSetting);
  Settings_IntSetting___ctor_40f39a0(pSVar1,0,-0x80000000,0x7fffffff,(MethodInfo *)0x0);
  (__this->fields).TitansKilledOther = pSVar1;
  il2cpp_runtime_helper_022b4080(&(__this->fields).TitansKilledOther);
  pSVar1 = (Settings_IntSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_IntSetting);
  Settings_IntSetting___ctor_40f39a0(pSVar1,0,-0x80000000,0x7fffffff,(MethodInfo *)0x0);
  (__this->fields).HumansKilledTotal = pSVar1;
  il2cpp_runtime_helper_022b4080(&(__this->fields).HumansKilledTotal);
  pSVar1 = (Settings_IntSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_IntSetting);
  Settings_IntSetting___ctor_40f39a0(pSVar1,0,-0x80000000,0x7fffffff,(MethodInfo *)0x0);
  (__this->fields).HumansKilledBlade = pSVar1;
  il2cpp_runtime_helper_022b4080(&(__this->fields).HumansKilledBlade);
  pSVar1 = (Settings_IntSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_IntSetting);
  Settings_IntSetting___ctor_40f39a0(pSVar1,0,-0x80000000,0x7fffffff,(MethodInfo *)0x0);
  (__this->fields).HumansKilledAHSS = pSVar1;
  il2cpp_runtime_helper_022b4080(&(__this->fields).HumansKilledAHSS);
  pSVar1 = (Settings_IntSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_IntSetting);
  Settings_IntSetting___ctor_40f39a0(pSVar1,0,-0x80000000,0x7fffffff,(MethodInfo *)0x0);
  (__this->fields).HumansKilledThunderspear = pSVar1;
  il2cpp_runtime_helper_022b4080(&(__this->fields).HumansKilledThunderspear);
  pSVar1 = (Settings_IntSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_IntSetting);
  Settings_IntSetting___ctor_40f39a0(pSVar1,0,-0x80000000,0x7fffffff,(MethodInfo *)0x0);
  (__this->fields).HumansKilledAPG = pSVar1;
  il2cpp_runtime_helper_022b4080(&(__this->fields).HumansKilledAPG);
  pSVar1 = (Settings_IntSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_IntSetting);
  Settings_IntSetting___ctor_40f39a0(pSVar1,0,-0x80000000,0x7fffffff,(MethodInfo *)0x0);
  (__this->fields).HumansKilledTitan = pSVar1;
  il2cpp_runtime_helper_022b4080(&(__this->fields).HumansKilledTitan);
  pSVar1 = (Settings_IntSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_IntSetting);
  Settings_IntSetting___ctor_40f39a0(pSVar1,0,-0x80000000,0x7fffffff,(MethodInfo *)0x0);
  (__this->fields).HumansKilledOther = pSVar1;
  il2cpp_runtime_helper_022b4080(&(__this->fields).HumansKilledOther);
  __this_00 = (GameProgress_DamageSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_DamageSetting);
  Settings_BaseSetting___ctor((Settings_BaseSetting_o *)__this_00,(MethodInfo *)0x0);
  (*(__this_00->klass->vtable)._4_SetDefault.methodPtr)
            (__this_00,(__this_00->klass->vtable)._4_SetDefault.method);
  (__this->fields).Damage = __this_00;
  il2cpp_runtime_helper_022b4080(&(__this->fields).Damage);
  Settings_BaseSettingsContainer___ctor((Settings_PresetSettingsContainer_o *)__this,(MethodInfo *)0x0);
  return;
}

File: csharp_body_deref/Scripts/GameProgress/GameStatContainer.cs
// ==================== AoTTG2 cross-reference ====================
// Type: GameProgress.GameStatContainer
// Update status: unchanged since the previous dump
// Real method bodies (native C decompile): source/cpp/GameProgress/GameStatContainer.c
// Prior real C# source (older reference): Assets/Scripts/GameProgress/GameStatContainer.cs
// Bodies below are empty IL2CPP metadata stubs; the logic lives in the native C.
// ===============================================================
using Il2CppDummyDll;
using Settings;

namespace GameProgress;

[Token(Token = "0x20001FB")]
internal class GameStatContainer : BaseSettingsContainer
{
	[Token(Token = "0x4000A97")]
	[FieldOffset(Offset = "0x20")]
	public IntSetting Level;

	[Token(Token = "0x4000A98")]
	[FieldOffset(Offset = "0x28")]
	public IntSetting Exp;

	[Token(Token = "0x4000A99")]
	[FieldOffset(Offset = "0x30")]
	public FloatSetting PlayTime;

	[Token(Token = "0x4000A9A")]
	[FieldOffset(Offset = "0x38")]
	public FloatSetting HighestSpeed;

	[Token(Token = "0x4000A9B")]
	[FieldOffset(Offset = "0x40")]
	public IntSetting TitansKilledTotal;

	[Token(Token = "0x4000A9C")]
	[FieldOffset(Offset = "0x48")]
	public IntSetting TitansKilledBlade;

	[Token(Token = "0x4000A9D")]
	[FieldOffset(Offset = "0x50")]
	public IntSetting TitansKilledAHSS;

	[Token(Token = "0x4000A9E")]
	[FieldOffset(Offset = "0x58")]
	public IntSetting TitansKilledThunderspear;

	[Token(Token = "0x4000A9F")]
	[FieldOffset(Offset = "0x60")]
	public IntSetting TitansKilledAPG;

	[Token(Token = "0x4000AA0")]
	[FieldOffset(Offset = "0x68")]
	public IntSetting TitansKilledOther;

	[Token(Token = "0x4000AA1")]
	[FieldOffset(Offset = "0x70")]
	public IntSetting HumansKilledTotal;

	[Token(Token = "0x4000AA2")]
	[FieldOffset(Offset = "0x78")]
	public IntSetting HumansKilledBlade;

	[Token(Token = "0x4000AA3")]
	[FieldOffset(Offset = "0x80")]
	public IntSetting HumansKilledAHSS;

	[Token(Token = "0x4000AA4")]
	[FieldOffset(Offset = "0x88")]
	public IntSetting HumansKilledThunderspear;

	[Token(Token = "0x4000AA5")]
	[FieldOffset(Offset = "0x90")]
	public IntSetting HumansKilledAPG;

	[Token(Token = "0x4000AA6")]
	[FieldOffset(Offset = "0x98")]
	public IntSetting HumansKilledTitan;

	[Token(Token = "0x4000AA7")]
	[FieldOffset(Offset = "0xA0")]
	public IntSetting HumansKilledOther;

	[Token(Token = "0x4000AA8")]
	[FieldOffset(Offset = "0xA8")]
	public DamageSetting Damage;

	[Token(Token = "0x6000BC6")]
	[Address(RVA = "0x443D5F0", Offset = "0x443D5F0", VA = "0x443D5F0")]
	public GameStatContainer()
	{
	}
}
